library/src/main/java/com/mapzen/android/graphics/OverlayManager.java

```diff
@@ -161,14 +161,18 @@ public void setMyLocationEnabled(boolean enabled) {
    * Optionally persists the map data to the {@link MapDataManager}.
    */
   public void setMyLocationEnabled(boolean enabled, boolean persistMapData) {
```

Comment:
I solved for this case by always removing the current location persisted map data when `persistMapData` is `true`. Then I only re-add that layer if `enabled` is also `true`.

Also took this chance to break persistence logic and Tangram map data UI logic into separate private methods.